src/shared_modules/indexer_connector/tests/component/fakeIndexer.hpp

```diff
@@ -0,0 +1,211 @@
+ * January 09, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _FAKE_INDEXER_HPP
+#define _FAKE_INDEXER_HPP
+
+#include <external/cpp-httplib/httplib.h>
+#include <external/nlohmann/json.hpp>
+#include <functional>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <utility>
+
+/**
+ * @brief This class is a simple HTTP server that provides a fake OpenSearch server.
+ */
+class FakeIndexer
+{
+private:
+    httplib::Server m_server;
+    std::thread m_thread;
+    int m_port;
+    std::string m_host, m_health, m_indexName;
+    bool m_indexerInitialized;
+    std::function<void(const std::string&)> m_initTemplateCallback = {};
+    std::function<void(const std::string&)> m_initIndexCallback = {};
+    std::function<void(const std::string&)> m_publishCallback = {};
+
+public:
+    /**
+     * @brief Class constructor.
+     *
+     * @param host Host of the server.
+     * @param port Port of the server.
+     * @param health Health status of the server.
+     * @param indexName Name of the index.
+     */
+    FakeIndexer(std::string host, int port, std::string health, std::string indexName)
+        : m_thread(&FakeIndexer::run, this)
+        , m_host(std::move(host))
+        , m_health(std::move(health))
+        , m_port(port)
+        , m_indexName(std::move(indexName))
+        , m_indexerInitialized(false)
+        , m_publishCallback({})
+        , m_initIndexCallback({})
+        , m_initTemplateCallback({})
+    {
+        // Wait until server is ready.
+        while (!m_server.is_running())
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        }
+    }
+
+    ~FakeIndexer()
+    {
+        m_server.stop();
+        if (m_thread.joinable())
+        {
+            m_thread.join();
+        }
+    }
+
+    /**
+     * @brief Sets the health of the server.
+     *
+     * @param health New health value.
+     */
+    void setHealth(std::string health)
+    {
+        m_health = health;
+    }
+
+    /**
+     * @brief Sets the publish callback.
+     *
+     * @param callback New callback.
+     */
+    void setPublishCallback(std::function<void(const std::string&)> callback)
+    {
+        m_publishCallback = callback;
+    }
+
+    /**
+     * @brief Sets the init index callback.
+     *
+     * @param callback New callback.
+     */
+    void setInitIndexCallback(std::function<void(const std::string&)> callback)
+    {
+        m_initIndexCallback = callback;
+    }
+
+    /**
+     * @brief Sets the init template callback.
+     *
+     * @param callback New callback.
+     */
+    void setInitTemplateCallback(std::function<void(const std::string&)> callback)
+    {
+        m_initTemplateCallback = callback;
+    }
+
+    /**
+     * @brief Returns a the initialized variable.
```

Comment:
Seems to be a typo

Also what do you think about this?
```suggestion
     * @brief Returns the indexer initialized flag.
```

Reply:
Fixed.